### PR TITLE
Restore FileAccess.close method.

### LINKED ISSUE #6301: Add FileAccess.close() method to GDScript and encourage its use in documentation and samples
### Describe the project you are working on

Several small demos and tutorials.

### Describe the problem or limitation you are having in your project

Explaining Godot 4 file handling to Godot 3 users, beginners and people coming from other engines.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

File and directory access were reworked in https://github.com/godotengine/godot/pull/65271, including removing the File.close()-method. So, instead of the old Godot 3 way:

``` gdscript
var f = File.new()
f.open(...)
...
f.close() #Closes the file.
```

In Godot 4, you now do:

``` gdscript
var f = FileAccess.open(...)
...
f = null #Closes the file (if no other references to it exist).
```

I prefer the old Godot 3 way of explicitly calling close(). Some random points I have encountered to support this proposal:

1) While functionally similar, I feel like this "nulling" out the reference to close a file is somewhat of an exception to the general rules of external OS-level resource handling in Godot. Most other similar resources have a close() method. For example ZIPPacker/Reader, WebSocketPeer, HTTPClient and many others have close()-methods while nodes and raw Objects need to be released using free() or queue_free().

2) In terms of communication, an explicit close() signals the programmer intent very well. Close the file and flush any buffers. Nulling the reference can mean other things. Why is it done here? Maybe it was a shared reference to some open log file we no longer need but is still used by other instances? Is there another reference alive somewhere? Does this actually flush and close the file yet? You need to understand the surrounding code well when you see this, with an explicit close() call the intent is more clear. This also makes sure any other place that still mistakenly reads or writes the file gives errors after closing the file, alerting the programmer that the file is still incorrectly used somewhere.

3) To people used to Python or some similar programming language, simply nulling an OS-level resource often looks like a beginner bug or misunderstanding made by someone who doesn't fully understand references and resource management. It is also discouraged and in developer mode Python warns if a file object is garbage collected without being explicitly closed (although the preferred way is to use context managers and `with`-keywords which handles that). They'll probably dig the Godot documentation and see that nulling idiom closes the file. Of course, the next time they'll use some other resource they might assume the same and now their code leaks objects.

4) There is a lot of GDScript code out there that leaks nodes (and other non-RefCounted Objects) because people think nulling all references to those objects is enough to free them. Let's not encourage GDScript beginners to write (and read) code patterns like that. Encourage people to explicitly manage their resource lifecycles with disposal calls when available.

5) Also, the most important thing: calling close() and other disposal methods is always extremely satisfactory. It feels final. It means your job is now done. You have survived and reached your goal. You can go home and relax. Your boss will praise you and give you a raise. The operating system and hardware will take the blame if something goes wrong after it.

6) There will be less room to suffer from bugs like this: https://github.com/godotengine/godot/issues/73355 and explicit closing also shields from future bugs like that. References can accidentally leak or stay alive longer than intended in surprising ways, from bugs in debugger variable watches to issues with stack frame storage. Not to forget simple programmer logic errors.

Note that adding close() will still keep nulling the reference as a valid alternative. This proposal does not break compatibility. Both options will work (just like in Godot 3), but I think many people would prefer to explicitly close() files. Also calling close() multiple times should probably be a no-op without a need for a warning or error message. And those who want to reach Zen with a total clarity and peace of mind can do both close() and nulling after it.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Add FileAccess.close() to explicitly close a file.

``` gdscript
var f = FileAccess.open(...)
...
f.close()
```

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It will be used often, file handling is important in most applications.

### Is there a reason why this should be core and not an add-on in the asset library?

FileAccess is core functionality.

### PR DESCRIPTION
**Pull request by bruvzg**
Closes https://github.com/godotengine/godot-proposals/issues/6301

### CODE REVIEW

**Comment by YuriSizov** [comment]
Project settings changes seem unrelated?

**Comment by bruvzg** [comment]
> Project settings changes seem unrelated?

I guess it's another platform dependent option which changes on docs update, should be changed to `_NOVAL` to ignore default value it.

**Review by akien-mga — APPROVED** [review]

File: doc/classes/FileAccess.xml [hunk 1]
```diff
@@ -52,6 +52,13 @@
 		<link title="3D Voxel Demo">https://godotengine.org/asset-library/asset/676</link>
 	</tutorials>
 	<methods>
+		<method name="close">
+			<return type="void" />
+			<description>
+				Closes the currently opened file and prevents subsequent read/write operations. Use flush to persist the data to disk without closing the file.
```
Comment (KoBeWi) [inline doc/classes/FileAccess.xml:58]:
```suggestion
				Closes the currently opened file and prevents subsequent read/write operations. Use [method flush] to persist the data to disk without closing the file.
```

**Comment by akien-mga** [comment]
Thanks!

File: doc/classes/FileAccess.xml (hunk 1 above)
Comment (akien-mga) [reply doc/classes/FileAccess.xml:58]: Forgot about this before merging. One of us can sneak this nitpick into another unrelated bugfix or documentation commit next time :P

**Pull request merged**